### PR TITLE
Remove Foodcritic in favor of Cookstyle

### PR DESCRIPTION
**Pull request by jakauppila**
Foodcritic has been deprecated in favor of [Cookstyle](https://docs.chef.io/workstation/cookstyle/), so removed any references to Foodcritic, added tests for Cookstyle, and auto-corrected all Cookstyle syntax issues.

### CODE REVIEW

**Comment by jakauppila** [comment]
@Annih I'm assuming there's some manual steps required to get these kitchen converges to fire?

image

**Comment by Annih** [comment]
As for windows we are using EC2 kitchen boxes, PR coming from external repositories have to deploy their own secrets; I'll validate your PR manually.

File: resources/framework.rb [hunk 1]
```diff
@@ -21,7 +21,8 @@
 #
 
 # The provides method is available on chef >= 12.0 only
-provides :ms_dotnet_framework, os: 'windows' if respond_to?(:provides)
+provides :ms_dotnet_framework, os: 'windows'
+unified_mode true if respond_to?(:provides)
```
Comment (Annih) [inline resources/framework.rb:25]: I think you mean `if respond_to?(:unified_mode)`

File: resources/reboot.rb [hunk 2]
```diff
@@ -21,7 +21,8 @@
 #
 
 # The provides method is available on chef >= 12.0 only
-provides :ms_dotnet_reboot, os: 'windows' if respond_to?(:provides)
+provides :ms_dotnet_reboot, os: 'windows'
+unified_mode true if respond_to?(:provides)
```
Comment (Annih) [inline resources/reboot.rb:25]: Same as in the other file: `if respond_to?(:unified_mode)`

**Review by Annih — CHANGES_REQUESTED** [review]
Thank you for this PR @jakauppila , there are 2 tiny thing in the resources, but apart from that the whole change is nice!

File: resources/framework.rb (hunk 1 above)
Comment (jakauppila) [reply resources/framework.rb]: Yup, fixed. Odd that happened since most of the changes here were generated by using autocorrect with Cookstyle

**Comment by jakauppila** [comment]
@Annih Fixed that issue. If you're interested I could cut a separate PR to move the kitchen executions to run via Github actions on the public runners.

**Comment by jakauppila** [comment]
@Annih Just wanted to ping on this again.

**Comment by Annih** [comment]
OK thank you @jakauppila for the change and sorry for the delay.

Feel free to push a new PR for the kitchen suite on public GA runners.

**Review by Annih — COMMENTED** [review]
ok

**Pull request merged**